### PR TITLE
Fix host animation for leave

### PR DESCRIPTION
**Pull request by tbosch**
**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[ ] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: packages/core/src/codegen_private_exports.ts [hunk 1]
```diff
@@ -10,4 +10,4 @@ export {CodegenComponentFactoryResolver as ɵCodegenComponentFactoryResolver} fr
...
+export {ArgumentType as ɵArgumentType, BindingFlags as ɵBindingType, DepFlags as ɵDepFlags, EMPTY_ARRAY as ɵEMPTY_ARRAY, EMPTY_MAP as ɵEMPTY_MAP, NodeFlags as ɵNodeFlags, QueryBindingType as ɵQueryBindingType, QueryValueType as ɵQueryValueType, ViewDefinition as ɵViewDefinition, ViewFlags as ɵViewFlags, anchorDef as ɵand, createComponentFactory as ɵccf, createRendererType2 as ɵcrt, directiveDef as ɵdid, elementDef as ɵeld, elementEventFullName as ɵelementEventFullName, getComponentViewDefinitionFactory as ɵgetComponentViewDefinitionFactory, inlineInterpolate as ɵinlineInterpolate, interpolate as ɵinterpolate, ngContentDef as ɵncd, nodeValue as ɵnov, pipeDef as ɵpid, providerDef as ɵprd, pureArrayDef as ɵpad, pureObjectDef as ɵpod, purePipeDef as ɵppd, queryDef as ɵqud, textDef as ɵted, unwrapValue as ɵunv, viewDef as ɵvid} from './view/index';
```
Comment (vicb) [inline packages/core/src/codegen_private_exports.ts]: `ɵBindingFlags`

File: packages/core/test/animation/animation_integration_spec.ts [hunk 2]
```diff
@@ -80,42 +80,207 @@ export function main() {
...
+      describe('host bindings', () => {
+        it('should trigger a state change animation from void => state on the component host element',
+           fakeAsync(() => {
+             @Component({
+               selector: 'my-cmp',
+               template: '...',
+               animations: [trigger(
+                   'myAnimation',
+                   [transition(
+                       'a => b',
```
Comment (vicb) [inline packages/core/test/animation/animation_integration_spec.ts]: `void =>smthg`

File: packages/core/test/animation/animation_integration_spec.ts [hunk 3]
```diff
@@ -80,42 +80,207 @@ export function main() {
...
+                       [style({'opacity': '0'}), animate(500, style({'opacity': '1'}))])])],
+             })
+             class Cmp {
+               @HostBinding('@myAnimation')
+               get binding() { return this.exp ? 'b' : 'a'; }
+
+               exp: any = false;
```
Comment (vicb) [inline packages/core/test/animation/animation_integration_spec.ts]: nit: boolean

File: packages/core/test/animation/animation_integration_spec.ts [hunk 4]
```diff
@@ -80,42 +80,207 @@ export function main() {
...
+             cmp.exp = true;
+             fixture.detectChanges();
+             engine.flush();
+             expect(getLog().length).toEqual(1);
+
+             const data = getLog().pop();
+             expect(data.element).toEqual(fixture.elementRef.nativeElement);
+             expect(data.keyframes).toEqual([{offset: 0, opacity: '0'}, {offset: 1, opacity: '1'}]);
+
+             flushMicrotasks();
```
Comment (vicb) [inline packages/core/test/animation/animation_integration_spec.ts]: no child, remove

**Review by vicb — APPROVED** [review]

**Comment by matsko** [comment]
I'm just commenting here to confirm that Tobias is including my testing commit so that we can guarantee that the animation tests now work.

File: packages/core/src/codegen_private_exports.ts (hunk 1 above)
Comment (tbosch) [inline packages/core/src/codegen_private_exports.ts]: removed

File: packages/core/test/animation/animation_integration_spec.ts (hunk 2 above)
Comment (tbosch) [inline packages/core/test/animation/animation_integration_spec.ts]: Changed description of the test.

File: packages/core/test/animation/animation_integration_spec.ts (hunk 3 above)
Comment (tbosch) [inline packages/core/test/animation/animation_integration_spec.ts]: done

File: packages/core/test/animation/animation_integration_spec.ts (hunk 4 above)
Comment (tbosch) [inline packages/core/test/animation/animation_integration_spec.ts]: Done

**Pull request closed without merging**